submissions/kt-std/friends-app/js/script.js

```diff
@@ -0,0 +1,338 @@
+}
+
+function getFriendCardTemplate(friend) {
+	return `<div class="card__container shadow">
+				<div class="card__row around">
+					<a href='mailto:${friend.email}' class="email__button button" 
+					   data-title='${friend.email}'></a>
+					<img src="${friend.image}" class="card__image">
+					<a href='tel:${reformatPhoneNumber(friend.phone)}' class="phone__button button" 
+					   data-title='${reformatPhoneNumber(friend.phone)}'></a>
+				</div>
+				<div class="card__row column">
+					<h3 class="card__name">${friend.firstName} ${friend.lastName}</h3>
+					<h5 class="card__username">@${friend.username}</h5>
+				</div>
+				<div class="card__row gender__container">
+					<h6 class="card__gender">${getGenderIcon(friend.gender)}</h6>
+					<h6 class="card__age">${friend.age}</h6>
+				</div>
+				<div class="card__row registered__container">
+					<p class="registered__message">Friends since <br> 
+					${getDate(friend.registeredDate)}</p>
+					<div style="width:${friend.registredAge}%"></div>
+				</div>
+				<div class="card__row country__row">
+					<h6 class="card__country">${friend.country}</h6>
+				</div>
+			</div>`;
+}
+
+function reformatPhoneNumber(number) {
+	return number
+		.replace(/[^0-9]+/g, "")
+		.replace(/.(\d{3})/g, "$1-")
+		.replace(/(^\d{3,3})-(.\d+)/, "+($1)-$2")
+		.replace(/[-]+$/g, "");
+}
+
+function getDate(date) {
+	return new Date(date).toLocaleString("en-US", {
+		year: "numeric",
+		month: "long",
+		day: "numeric",
+	});
+}
+
+function getGenderIcon(gender) {
+	return gender === "female"
+		? '<span class="female">♀</span>'
+		: '<span class="male">♂</span>';
+}
+
+function getResponseErrorMessage(status, statusText) {
+	return `<h1 class='error__message'>Sorry, an error occured!</h1>
+			<h2 class='error__code'>${status}: ${statusText}</h2>`;
+}
+
+function setAgeLimit(ageLimit, limitInputId, limitType) {
+	const limitInput = document.getElementById(limitInputId);
+	limitInput[limitType] = ageLimit;
+	setInputLimitValue(
+		limitInput,
+		limitInputId,
+		limitType,
+		"minAge",
+		"min",
+		ageLimit
+	);
+	setInputLimitValue(
+		limitInput,
+		limitInputId,
+		limitType,
+		"maxAge",
+		"max",
+		ageLimit
+	);
+}
+
+function setInputLimitValue(
+	limitInput,
+	limitInputId,
+	limitType,
+	limitInputIdValue,
+	limitTypeValue,
+	ageLimit
+) {
+	if (limitInputId === limitInputIdValue && limitType === limitTypeValue)
+		limitInput.value = ageLimit;
+}
+
+function getCertainAgeLimit(friendsArray, value) {
+	const sortedArray = friendsArray.sort((a, b) => a.age - b.age);
+	return value === "min"
+		? sortedArray[0].age
+		: sortedArray[sortedArray.length - 1].age;
+}
+
+function sortCards(e, friendsArray) {
+	if (e.target.classList.contains("list__item")) {
+		document.querySelector(".select__face-item").innerText =
+			e.target.textContent;
+		sortCardsArray(e.target.attributes.value.value, friendsArray);
+	}
+}
+
+function sortCardsArray(condition, friendsArray) {
+	switch (condition) {
+		case "ND":
```

Comment:
`ND` and `NA` is something from ninja code :) https://learn.javascript.ru/ninja-code

Reply:
I'm afraid that I'm incapable of achieving such perfection :satisfied: Although, most principles are followed. Unfortunately:)
I've updated it with human-readable values.